storage/spider/CMakeLists.txt

```diff
@@ -50,11 +50,8 @@ ELSEIF(PLUGIN_PARTITION MATCHES "^NO$")
 ELSE()
   INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/storage/spider/hs_client)
 
-  INSTALL(FILES
-    ${CMAKE_SOURCE_DIR}/storage/spider/scripts/install_spider.sql
-    DESTINATION ${INSTALL_MYSQLSHAREDIR} COMPONENT Server
-  )
-  MYSQL_ADD_PLUGIN(spider ${SPIDER_SOURCES} STORAGE_ENGINE MODULE_ONLY MODULE_OUTPUT_NAME "ha_spider")
+  MYSQL_ADD_PLUGIN(spider ${SPIDER_SOURCES}
+    STORAGE_ENGINE COMPONENT spider-engine MODULE_ONLY MODULE_OUTPUT_NAME "ha_spider")
```

Comment:
???
Why did you need to specify a component here?
In rpm builds spider should go into the server rpm.

Reply:
ok, you need a component to have a generated .cnf file.
this is wrong, spider still should go into the server rpm.

Reply:
This was already merged, so I will not start reverting anything at this point.

I checked if further changes are needed and drafted #1586 but after checking everything, current 10.5 branch with this PR and one of your post-fix commits seems to cover everything, thus no further changes needed for now.